File: reverseInteger.py
"""
Reverse digits of an integer.

Example1: x = 123, return 321
Example2: x = -123, return -321
"""

class Solution:
    # @return an integer
    def reverse(self, x):
    	result = 0
    	flag = 1
    	if x < 0:
    		flag = -1
    		x = abs(x)
        while x > 0:
        	result *= 10
        	result += x % 10
        	x /= 10
        return flag * result

s = Solution()
print s.reverse(-1065478)

File: JAVA/reverseInteger.java
public class Solution {
    public int reverse(int x) {
        int sign = 1, reverse = 0;
        if (x < 0) {
            sign = -1; 
            x *= -1;
        }
        while (x > 0){
            reverse *= 10;
            reverse += x % 10;
            x /= 10;
        }
        return reverse*sign;
    }
}